modules/seqexec/server/src/main/scala/seqexec/server/SeqTranslate.scala

```diff
@@ -365,75 +360,63 @@ class SeqTranslate(site: Site, systems: Systems[IO], settings: TranslateSettings
-    val tcsKReader = if (settings.tcsKeywords) TcsKeywordsReaderEpics[IO](TcsEpics.instance) else DummyTcsKeywordsReader[IO]
     sys.resource match {
       case Instrument.F2     =>
-        Flamingos2Header.header[IO](sys, Flamingos2Header.ObsKeywordsReaderODB(config), tcsKReader).asRight
+        Flamingos2Header.header[IO](sys, Flamingos2Header.ObsKeywordsReaderODB(config), systems.tcsKeywordReader).asRight
       case Instrument.GmosS |
            Instrument.GmosN  =>
-        val gmosInstReader = if (settings.gmosKeywords) GmosKeywordReaderEpics[IO](GmosEpics.instance) else GmosKeywordReaderDummy[IO]
-        GmosHeader.header[IO](sys, GmosObsKeywordsReader(config), gmosInstReader, tcsKReader).asRight
+        GmosHeader.header[IO](sys, GmosObsKeywordsReader(config), systems.gmosKeywordReader, systems.tcsKeywordReader).asRight
       case Instrument.Gnirs  =>
-        val gnirsReader = if(settings.gnirsKeywords) GnirsKeywordReaderEpics[IO](GnirsEpics.instance) else GnirsKeywordReaderDummy[IO]
-        GnirsHeader.header[IO](sys, gnirsReader, tcsKReader).asRight
+        GnirsHeader.header[IO](sys, systems.gnirsKeywordReader, systems.tcsKeywordReader).asRight
       case Instrument.Gpi    =>
-        GpiHeader.header[IO](systems.gpi.gdsClient, tcsKReader, ObsKeywordReader[IO](config, site)).asRight
+        GpiHeader.header[IO](systems.gpi.gdsClient, systems.tcsKeywordReader, ObsKeywordReader[IO](config, site)).asRight
       case Instrument.Ghost  =>
         GhostHeader.header[IO].asRight
       case Instrument.Niri   =>
-        val niriReader = if(settings.niriKeywords) NiriKeywordReaderEpics[IO](NiriEpics.instance)
-                          else NiriKeywordReaderDummy[IO]
-        NiriHeader.header[IO](sys, niriReader, tcsKReader).asRight
+        NiriHeader.header[IO](sys, systems.niriKeywordReader, systems.tcsKeywordReader).asRight
       case Instrument.Nifs   =>
-        val nifsReader = if(settings.nifsKeywords) NifsKeywordReaderEpics[IO](NifsEpics.instance) else NifsKeywordReaderDummy[IO]
-        NifsHeader.header[IO](sys, nifsReader, tcsKReader).asRight
+        NifsHeader.header[IO](sys, systems.nifsKeywordReader, systems.tcsKeywordReader).asRight
       case Instrument.Gsaoi   =>
-        val gsaoiReader = if (settings.gsaoiKeywords) GsaoiKeywordReaderEpics[IO](GsaoiEpics.instance) else GsaoiKeywordReaderDummy[IO]
-        GsaoiHeader.header[IO](sys, tcsKReader, gsaoiReader).asRight
+        GsaoiHeader.header[IO](sys, systems.tcsKeywordReader, systems.gsaoiKeywordReader).asRight
       case _                 =>
         TrySeq.fail(Unexpected(s"Instrument ${sys.resource} not supported."))
     }
   }
 
-  private def commonHeaders[F[_]: Sync](epics: => TcsEpics[F], config: CleanConfig, tcsSubsystems: List[TcsController.Subsystem],
-                            inst: InstrumentSystem[F])(ctx: HeaderExtraData): Header[F] =
+  private def commonHeaders(config: CleanConfig, tcsSubsystems: List[TcsController.Subsystem],
```

Comment:
Why are we going back to `IO`?

Reply:
I was forced because the keyword readers now come in `systems`, which is `System[IO]`. But don't worry, we'll get back the `F` in my next PR.

Reply:
👍